tests/acceptance/features/bootstrap/SettingsContext.php

```diff
@@ -153,15 +153,37 @@ public function theAdministratorHasGivenUserTheRole(string $user, string $role):
 	 * @return string
 	 */
 	public function getRoleIdByRoleName(string $user, string $role): string {
-		$response = $this->getRoles($user);
-		$this->featureContext->theHTTPStatusCodeShouldBe(
-			201,
-			"Expected response status code should be 201",
-			$response
-		);
+		// Sometimes the response body is not complete and results invalid json.
+		// so we try again until we get a valid json.
+		$tryAgain = false;
+		$retried = 0;
+		do {
+			$response = $this->getRoles($user);
+			$this->featureContext->theHTTPStatusCodeShouldBe(
+				201,
+				"Expected response status code should be 201",
+				$response
+			);
+
+			$rawBody =  $response->getBody()->getContents();
+			try {
+				$decodedBody = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
do we need `$tryAgain = false` or `return` statement inside the loop so we don't get inside the inifinite loop if $tryAgain is `true` ?